### PR TITLE
fix(ci): Fix clippy

### PR DESCRIPTION
**Pull request by last-partizan**
This is attempt to fix CI pipeline

## What kind of change does this PR introduce?
- [x] Fix
- [ ] Feature
- [ ] Codestyle
- [ ] Refactor
- [ ] Other

## Did this PR introduce a breaking change?
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- [ ] Yes, please list breaking changes
- [x] No

### CODE REVIEW

**Comment by last-partizan** [comment]
@smolck please take a look at this.

I attempted to fix CI, and looks like it working.

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -15,21 +15,11 @@ jobs:
...
-      run: |
-        cargo fmt --all -- --check
-
-    - name: Lint with Clippy
-      uses: actions-rs/clippy-check@v1.0.7
-      with:
-        token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/checkout@v2
+      - run: rustup component add clippy rustfmt
+      - uses: actions-rs/clippy-check@v1
```
Comment (smolck) [inline .github/workflows/clippy.yml:20]: Not, but maybe add names for the checkout and clippy steps? Just makes things a bit cleaner/easier to read I think.

Other than that, LGTM.

**Comment by last-partizan** [comment]
Btw, what are clippy doing? it downloading all dependencies just to check our code?

Hmm, maybe this actually has sense, when language is typed. I just too used to python.

image

**Comment by smolck** [comment]
LGTM, squash your commits and I’ll merge it.

**Comment by smolck** [comment]
> Btw, what are clippy doing? it downloading all dependencies just to check our code?
> 
> Hmm, maybe this actually has sense, when language is typed. I just too used to python.

Yeah I think it needs to compile the project and then it checks/lints it for various things defined in clippy.

**Comment by last-partizan** [comment]
Squashed and merged :)

**Pull request merged**